### PR TITLE
expanding available bidders chart stats

### PR DESCRIPTION
**Pull request by SidelineCory24**
The request was to add the ability to see the Skill, Grade, Location, Bureau/Post (preferably Bureau), and TED to the chart, in addition to the Status.

Test with this [BE PR](https://github.com/MetaPhase-Consulting/State-TalentMAP-API/pull/604)

- [x] update Grade to real data on BE
- [x] move FE data to BE (including colors)
- [x] dynamically grab values (grade first)
- [x] add other dropdown options
- [x] find a color randomizer (contrasting)
- [x] adjust css for larger stats columns

### CODE REVIEW

**Comment by SidelineCory24** [comment]
Initial Grade testing (hard coded data)
image

File: src/Components/AvailableBidder/AvailableBidderStats/AvailableBidderStats.jsx [hunk 1]
```diff
@@ -1,44 +1,44 @@
...
-
-  data = data.map(m => ({ ...m, value: get(stats, m.key, 0) }));
+  const stats = get(biddersData.stats, selectedStat, []);
+  const statsSum = get(biddersData.stats, 'Sum', {})[selectedStat];
 
-  const data$ = data.map(m => {
-    // handling division by zero
-    const sum = statsSum !== 0 ? statsSum : 1;
+  // adding colors
+  const stats$ = stats.map(m => {
```
Comment (SidelineCory24) [inline src/Components/AvailableBidder/AvailableBidderStats/AvailableBidderStats.jsx:33]: The color of the slices will change each time as you select different options in the dropdown.

Comment (elizabeth-jimenez) [reply src/Components/AvailableBidder/AvailableBidderStats/AvailableBidderStats.jsx:33]: @scott062 @mjoyce91 are y'all cool with this? Personally don't mind it (if we do want more constant colors, Cory had a solution already implemented on the BE that could be pulled back in)

File: src/Components/AvailableBidder/AvailableBidderStats/AvailableBidderStats.jsx [hunk 2]
```diff
@@ -1,44 +1,44 @@
...
+    const color = getAvatarColor((m.name), 6000).backgroundColor;
     return {
       ...m,
-      percent: numeral(m.value / sum).format('0%'),
+      color,
     };
   });
 
-  const chartData$ = data$.filter(f => f.value > 0);
+  const legendClass = stats$.length > 40 ? 'legend-large' : 'legend-small';
```
Comment (elizabeth-jimenez) [inline src/Components/AvailableBidder/AvailableBidderStats/AvailableBidderStats.jsx:41]: Wow! ;)

File: src/Components/AvailableBidder/AvailableBidderStats/AvailableBidderStats.jsx [hunk 3]
```diff
@@ -58,20 +58,35 @@ const AvailableBidderStats = () => {
...
+                !availableBiddersIsLoading && !statsSum && `There are no available bidders categorized by ${selectedStat}.`
               }
               {
                 !availableBiddersIsLoading && !!statsSum &&
                 <div className="usa-grid-full flex">
-                  <div className="usa-width-one-fourth legend-container">
-                    <div className="usa-grid-full legend">
-                      <h4>Available Bidders by Status ({statsSum})</h4>
+                  <div className="legend-container">
+                    <h4>Available Bidders {selectedStat} Stats ({statsSum})</h4>
```
Comment (elizabeth-jimenez) [inline src/Components/AvailableBidder/AvailableBidderStats/AvailableBidderStats.jsx:86]: I wonder if we should change 'Stats' to 'Statistics'? If we are in need of space, we may also be able to drop "Available Bidders"

File: src/Components/AvailableBidder/AvailableBidderStats/AvailableBidderStats.jsx [hunk 4]
```diff
@@ -58,20 +58,35 @@ const AvailableBidderStats = () => {
             {
               showMore &&
             <div className="usa-grid-full section statistics-section">
+              {
+                <div className="filter-div">
+                  <div className="label">Chart Stats:</div>
```
Comment (elizabeth-jimenez) [inline src/Components/AvailableBidder/AvailableBidderStats/AvailableBidderStats.jsx:63]: same as https://github.com/MetaPhase-Consulting/State-TalentMAP/pull/1752#discussion_r745625655

File: src/Components/AvailableBidder/AvailableBidderStats/AvailableBidderStats.jsx [hunk 5]
```diff
@@ -1,44 +1,44 @@
...
-
-  let data = [
-    { name: 'Overcompliment (OC)', key: 'OC', value: 0, color: '#112E51' },
-    { name: 'Unassigned (UA)', key: 'UA', value: 0, color: '#205493' },
-    { name: 'In Transit (IT)', key: 'IT', value: 0, color: '#9BDAF1' },
-    { name: 'Absent without leave (AWOL)', key: 'AWOL', value: 0, color: '#02BFE7' },
-  ];
-
-  data = data.map(m => ({ ...m, value: get(stats, m.key, 0) }));
+  const stats = get(biddersData.stats, selectedStat, []);
```
Comment (elizabeth-jimenez) [inline src/Components/AvailableBidder/AvailableBidderStats/AvailableBidderStats.jsx:29]: `const stats = get(biddersData.stats, selectedStat) || [];` will catch when the key exists but is falsey

File: src/Components/AvailableBidder/AvailableBidderStats/AvailableBidderStats.jsx (hunk 3 above)
Comment (SidelineCory24) [reply src/Components/AvailableBidder/AvailableBidderStats/AvailableBidderStats.jsx:86]: There's enough room with us starting the columns flushed to the left now
image

File: src/Components/AvailableBidder/AvailableBidderStats/AvailableBidderStats.jsx (hunk 4 above)
Comment (SidelineCory24) [reply src/Components/AvailableBidder/AvailableBidderStats/AvailableBidderStats.jsx:63]: done in above ss

File: src/Components/AvailableBidder/AvailableBidderStats/AvailableBidderStats.jsx (hunk 5 above)
Comment (SidelineCory24) [reply src/Components/AvailableBidder/AvailableBidderStats/AvailableBidderStats.jsx:29]: done

File: src/Components/AvailableBidder/AvailableBidderStats/AvailableBidderStats.jsx (hunk 1 above)
Comment (SidelineCory24) [reply src/Components/AvailableBidder/AvailableBidderStats/AvailableBidderStats.jsx:33]: Robin said she didn't really care either about the colors change. No feedback from Chris yet. 

If needed we can change this on the other ticket I created yesterday with the suggestions we got from Robin.

Comment (elizabeth-jimenez) [reply src/Components/AvailableBidder/AvailableBidderStats/AvailableBidderStats.jsx:33]: Great idea!

**Comment by SidelineCory24** [comment]
can we still merge with the codeclimate/diff-coverage issue?

**Review by mjoyce91 — APPROVED** [review]

**Comment by mjoyce91** [comment]
@SidelineCory24 @elizabeth-jimenez - for this one yes, but in general let's start trying to hit the test coverage requirements

**Comment by SidelineCory24** [comment]
> @SidelineCory24 @elizabeth-jimenez - for this one yes, but in general let's start trying to hit the test coverage requirements

Ok sounds good

**Pull request merged**